docs/source/integrations/working_with_sql.md

```diff
@@ -0,0 +1,190 @@
+
+# Working with SQL in Kedro
+
+This page outlines the various approaches and recommended practices when building a Kedro pipeline with SQL databases.
+
+## Key concepts and assumptions
+
+When working with SQL databases in Kedro, it's helpful to keep the following concepts front of mind.
+
+### 1. Storage vs. Execution
+
+- You can treat a SQL database purely as a data sink/source, or
+- Leverage the database engine’s compute power whenever possible.
+
+### 2. CRUD Scope
+
+- Kedro handles the parts of CRUD which do not require some concept of “state.”
+- Kedro’s genesis centred around reproducible science, not operational concepts like [backfills](https://www.ssp.sh/brain/backfill/).
+
+| Operation | Supported? | Notes                                                   |
+|-----------|------------|---------------------------------------------------------|
+| Create    | ✅          | Writing new tables (or views)                           |
+| Read      | ✅          | Loading data into DataFrames or Ibis expressions        |
+| Update    | ❌          | Requires custom SQL outside Kedro’s DAG; breaks lineage |
+| Delete    | ❌          | Same as Update—avoid for full reproducibility           |
+
+### 3. DataFrame-centric
+
+- Kedro’s DAG and built-in datasets are optimised around Python DataFrames.
+- Embedding complex SQL scripts reduces visibility into dependencies and hinders reproducibility. See [Delta Lake integration guide](./deltalake_versioning.md) for more on handling `UPSERT`/`MERGE` like operations.
+
+### 4. Reproducibility & injection risk
+
+- Raw SQL strings can bypass Kedro’s topological sorting and introduce SQL-injection vulnerabilities.
+- Prefer testable, Pythonic expression APIs (Ibis, Spark) over raw SQL queries.
+
+## Approaches
+
+| Approach       | Status    | Setup Complexity   |
+|----------------|--------|--------------------|
+| **Pandas SQL** | Legacy | Minimal            |
+| **Spark-JDBC** | Legacy | High (JVM/cluster) |
+| **Ibis**       | Modern | Minimal                |
+
+---
+
+### 1. Pandas SQL (legacy)
+
+```yaml
+events_data:
+  type: pandas.SQLQueryDataSet
+  sql: SELECT * FROM events WHERE date > '2025-01-01'
+  credentials: sql_creds
+```
+
+- **Mechanism**
+  Wraps `pd.read_sql_query/table` and `pd.DataFrame.to_sql`.
+- **Pros**
+
+  - No extra services or JVM
+  - Pandas-native
+- **Cons**
+
+  - Loads full tables in memory
+  - Bypasses database optimizations
+  - Hard to parameterise/test
+  - Potential cost/security on large dumps
+
+---
+
+### 2. Spark-JDBC (legacy)
+
+```yaml
+sales_data:
+  type: spark.JDBCDataset
+  table: sales
+  credentials: jdbc_creds
+```
+
+- **Mechanism**
+  Spark’s DataFrame API over JDBC. Leverages predicate pushdown so filters/projections occur in-database.
```

Comment:
Again, there's a question of who this is written for. It does feel more like notes at this time; I think predicate pushdown is definitely something that requires at least some level of knowledge to know of (which is not all Kedro users)